vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,175 @@
+                "imageEvent");
+            publisher = _eventLoopProvider->getPublisher<kpsr::vision_ocv::ImageData>("imageEvent",
+                                                                                      poolSize,
+                                                                                      buildFunction,
+                                                                                      _cloneFunction);
+        } else {
+            _eventEmitterProvider.reset(
+                new kpsr::EventEmitterMiddlewareProvider<kpsr::vision_ocv::ImageData>(
+                    nullptr, "img", poolSize, buildFunction, _cloneFunction));
+
+            subscriber = _eventEmitterProvider->getSubscriber();
+            publisher = _eventEmitterProvider->getPublisher();
+        }
+    }
+
+    void TearDown(const ::benchmark::State &state)
+    {
+        subscriber->removeListener("receiveIm");
+        bool useEventLoop = state.range(1);
+        if (useEventLoop) {
+            _eventLoopProvider->stop();
+        }
+    }
+
+    void setListener(benchmark::State &state)
+    {
+        subscriber
+            ->registerListener("receiveIm", [&state, this](const kpsr::vision_ocv::ImageData &im) {
+                auto now = std::chrono::duration_cast<std::chrono::milliseconds>(
+                               std::chrono::system_clock::now().time_since_epoch())
+                               .count();
+                state.SetIterationTime(now - im.timestamp);
+                eventsListened++;
+            });
+    }
+    int width, height;
+    kpsr::Subscriber<kpsr::vision_ocv::ImageData> *subscriber;
+    kpsr::Publisher<kpsr::vision_ocv::ImageData> *publisher;
+    std::atomic<int> eventsListened; // To be removed when pub/sub params will be implemented
+    std::function<void(kpsr::vision_ocv::ImageData &)> buildFunction;
+
+private:
+    std::unique_ptr<kpsr::vision_ocv::ImageDataFactory> _imageDataFactory;
+    std::function<void(const kpsr::vision_ocv::ImageData &, kpsr::vision_ocv::ImageData &)>
+        _cloneFunction;
+    std::unique_ptr<kpsr::EventEmitterMiddlewareProvider<kpsr::vision_ocv::ImageData>>
+        _eventEmitterProvider;
+    std::unique_ptr<kpsr::high_performance::EventLoopMiddlewareProvider<16>> _eventLoopProvider;
+};
+
+BENCHMARK_DEFINE_F(FactoryPerformanceTest, SimpleTest)(benchmark::State &state)
+{
+    bool useEventLoop = state.range(1);
+    setListener(state);
+
+    int eventsPublished = 0;
+
+    kpsr::vision_ocv::ImageData event;
+    buildFunction(event);
+    for (auto _ : state) {
+        event.timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+                              std::chrono::system_clock::now().time_since_epoch())
+                              .count();
+        publisher->publish(event);
+        eventsPublished++;
+        while (useEventLoop && eventsListened.load() < eventsPublished) {
```

Comment:
```suggestion
        while (useEventLoop && (eventsListened.load() < eventsPublished)) {
```
Having these parenthesis reduces ambiguity

Reply:
also, add another failsafe to prevent inifite waits in case of discarded events. Something like 

```
int failSafeCounter = 0;
const int failSafeCounterMax = 10;
while (useEventLoop && (eventsListened.load() < eventsPublished) && (failSafeCounter++ < failSafeCounterMax)) {
```